src/plugins/data/common/es_query/filters/get_display_value.ts

```diff
@@ -27,7 +28,16 @@ function getValueFormatter(indexPattern?: IIndexPattern, key?: string) {
   let format = get(indexPattern, ['fields', 'byName', key, 'format']);
   if (!format && (indexPattern.fields as any).getByName) {
     // TODO: Why is indexPatterns sometimes a map and sometimes an array?
-    format = ((indexPattern.fields as any).getByName(key) as IFieldType).format;
+    const field: IFieldType = (indexPattern.fields as any).getByName(key);
+    if (!field) {
+      throw new Error(
+        i18n.translate('data.filter.filterBar.fieldNotFound', {
+          defaultMessage: 'Field {key} not found',
```

Comment:
nit: It looks like we have the information to show `Field {field} not found in index pattern {index pattern}` to make troubleshooting the problem easier.